lib/robots/dor_repo/ocr/start_ocr.rb

```diff
@@ -11,9 +11,26 @@ def initialize
 
         def perform_work
           raise 'Object is already open' if object_client.version.status.open?
-          raise 'No files available or invalid object for OCR' unless Dor::TextExtraction::Ocr.new(cocina_object:).possible?
 
-          object_client.version.open(description: 'Start OCR workflow')
+          if Dor::TextExtraction::Ocr.new(cocina_object:).possible?
+            object_client.version.open(description: 'Start OCR workflow')
+          else
+            skip_workflow
+          end
+        end
+
+        private
+
+        # skip entire ocrWF workflow without opening version if OCR is not possible for this object
+        def skip_workflow
+          workflow_name = 'ocrWF'
+          status = 'skipped'
+          note = 'No files available or invalid object for OCR'
+          workflow = workflow_service.workflow(pid: druid, workflow_name:)
+          # get all of the incomplete steps, except for ourself (start-ocr), we will set that with the usual ReturnState
+          incomplete_steps = workflow.incomplete_processes.map(&:name).reject { |step| step == 'start-ocr' }
+          incomplete_steps.each { |process| workflow_service.update_status(druid:, workflow: workflow_name, process:, status:, note:) }
+          LyberCore::ReturnState.new(status: status.to_sym, note:)
```

Comment:
Yes, that's a good idea.  I'll make a ticket for this for future work, referencing this PR.  It would be fairly straightforward to implement if we need it again.